services/analyzeDuration.js

```diff
@@ -0,0 +1,73 @@
+const path = require("path");
+const fs = require("fs").promises;
+
+const mainFrameDirectory = path.join(__dirname, "../temp/frames/main-contents");
+const select1fpsFrames = require("../util/select1fpsFrames");
+
+const MOVEMENT_THRESHOLD = 0.09;
+
+async function analyzeDuration(
+  singleShotNumberList,
+  faceCountResults,
+  movementResults,
+) {
+  const mainFrameFileList = await fs.readdir(mainFrameDirectory);
+  const mainFramePathList = mainFrameFileList.map((frame) =>
+    path.join(mainFrameDirectory, frame),
+  );
+
+  mainFramePathList.sort((a, b) => {
+    const currentFramePath = parseInt(path.basename(a).split("_").pop(), 10);
+    const nextFramePath = parseInt(path.basename(b).split("_").pop(), 10);
+
+    return currentFramePath - nextFramePath;
+  });
+
+  const filteredFramePathList = select1fpsFrames(mainFramePathList);
+
+  const mainSingleShotPathList = filteredFramePathList.filter((framePath) => {
+    const frameNumber = parseInt(framePath.split("_").pop(), 10);
+
+    return singleShotNumberList.includes(frameNumber);
+  });
+
+  const replacementDuration = {};
+
+  for (
+    let singleShotIndex = 0;
+    singleShotIndex < mainSingleShotPathList.length - 1;
+    singleShotIndex += 1
+  ) {
+    let duration = 0;
+    const currentSingleShotPath = mainSingleShotPathList[singleShotIndex];
+    const nextSingleShotpath = mainSingleShotPathList[singleShotIndex + 1];
+
+    const currentSingleShotIndex = filteredFramePathList.findIndex(
+      (framePath) => framePath === currentSingleShotPath,
+    );
+    const nextSingleShotIndex = filteredFramePathList.findIndex(
+      (framePath) => framePath === nextSingleShotpath,
+    );
+
+    for (
+      let nextFrame = currentSingleShotIndex + 1;
+      nextFrame < nextSingleShotIndex;
+      nextFrame += 1
+    ) {
+      const { predictions } = faceCountResults[nextFrame];
+      const { movementRatio } = movementResults[nextFrame];
+
+      if (predictions.length >= 1 || movementRatio > MOVEMENT_THRESHOLD) {
+        duration += 1;
+      } else {
+        break;
+      }
```

Comment:
if else 문을 쓰는 것보다 if 문만 쓰는 방식은 어떻게 생각하시나요?!
```suggestion
      if (predictions.length < 1 && movementRatio <= MOVEMENT_THRESHOLD) {
        break;
      } 
      
       duration += 1;
```

Reply:
민석님! 해당 부분 반영하도록 하겠습니다! if문 하나로 처리하는 것이 가독성이 좋아서 고민하던 부분이었는데 감사합니다👍